core/src/main/java/org/infinispan/transaction/xa/CacheTransaction.java

```diff
@@ -144,4 +149,20 @@
     * Prevent new modifications after prepare or commit started.
     */
    void freezeModifications();
+
+
+   /**
+    * It returns a {@link CompletableFuture} that completes when the lock for the {@code key} is released.
+    *
+    * If the {@code key} is not locked by this transaction, it returns {@code null}.
+    *
+    * @param key the key.
+    * @return the {@link CompletableFuture} or {@link null} if the key is not locked by this transaction.
+    */
+   CompletableFuture<Void> getReleaseFutureForKey(Object key);
```

Comment:
Isn't this over-complicating things? `DefaultPendingLockManager` already knows that this transaction has the backup lock when calling this method, and we release all the backup locks at once, so we could use a single `CompletableFuture` for all the backup locks.